phoenix-core-client/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -940,8 +940,13 @@ private boolean addColumnsAndIndexesFromAncestors(MetaDataMutationResult result,
                 if (!alwaysAddAncestorColumnsAndIndexes && !result.wasUpdated() && !parentResult.wasUpdated()) {
                     return false;
                 }
-                result.setTable(ViewUtil.addDerivedColumnsAndIndexesFromParent(
-                        connection, table, parentTable));
+
+                PTable pTableWithDerivedColumnsAndIndexes
```

Comment:
You can remove the hasIndexId check from line 908 and use that here something like `if (!index || hasIndexId) call ViewUtil.addDerivedColumnsAndIndexesFromParent`

Reply:
Made this change, it was required to make sure the parent of an index table was present in cache so that we can add the ancestor map to the index.